lib/vector-vrl-functions/src/set_semantic_meaning.rs

```diff
@@ -60,7 +60,8 @@ impl Function for SetSemanticMeaning {
 
         let meaning = arguments
             .required_literal("meaning")?
-            .to_value()
+            .as_value()
+            .unwrap()
```

Comment:
I've noticed you've moved from `to_value()` to `as_value().unwrap()` in a couple of places in this PR. Could you elaborate on this change (why is it needed, why will it not panic)?

Reply:
Hmm, yeah. I should probably revert this.

`as_value` is a method on the `Expression` trait, which returns `Option<Value>`, and allows const expressions to return their value at compile-time.

`to_value` is a method implemented on the `Literal` expression, which forgoes the `Option`, since a literal is always a constant. I wanted to be consistent and always use `as_value`, but that means the need to unwrap, when we know its a literal expression.

I'll fix this up before I merge the PR.

Reply:
Resolved in [4e239c8](https://github.com/vectordotdev/vector/pull/12620/commits/4e239c8dd6190c4511f583aafd2ecd3de7c94334).